examples/cpp/csm/csm_greeter_client.cc

```diff
@@ -0,0 +1,205 @@
+    ClientContext context;
+
+    // The actual RPC.
+    std::mutex mu;
+    std::condition_variable cv;
+    bool done = false;
+    Status status;
+    if (cookieToServer != NULL) {
+      std::pair<std::string, std::string> cookieHeader =
+          cookieToServer->Header();
+      context.AddMetadata(cookieHeader.first, cookieHeader.second);
+    }
+    stub_->async()->SayHello(&context, &request, &reply,
+                             [&mu, &cv, &done, &status](Status s) {
+                               status = std::move(s);
+                               std::lock_guard<std::mutex> lock(mu);
+                               done = true;
+                               cv.notify_one();
+                             });
+
+    std::unique_lock<std::mutex> lock(mu);
+    while (!done) {
+      cv.wait(lock);
+    }
+
+    // Act upon its status.
+    if (status.ok()) {
+      if (cookieFromServer != NULL) {
+        const std::multimap<grpc::string_ref, grpc::string_ref>&
+            server_initial_metadata = context.GetServerInitialMetadata();
+        std::vector<Cookie> cookies =
+            GetCookies(server_initial_metadata, "GSSA");
+        if (!cookies.empty()) {
+          *cookieFromServer = cookies.front();
+        }
+      }
+      return reply.message();
+    } else {
+      std::cout << status.error_code() << ": " << status.error_message()
+                << std::endl;
+      return "RPC failed";
+    }
+  }
+
+ private:
+  std::unique_ptr<Greeter::Stub> stub_;
+};
+
+static void sayHello(GreeterClient& greeter, Cookie* cookieFromServer,
+                     const Cookie* cookieToServer) {
+  std::string user("world");
+  std::string reply = greeter.SayHello(user, cookieFromServer, cookieToServer);
+  std::cout << "Greeter received: " << reply << std::endl;
+  std::this_thread::sleep_for(std::chrono::seconds(5));
+}
+
+int main(int argc, char** argv) {
+  absl::ParseCommandLine(argc, argv);
+  opentelemetry::exporter::metrics::PrometheusExporterOptions opts;
+  // default was "localhost:9464" which causes connection issue across GKE pods
+  opts.url = "0.0.0.0:9464";
+  auto prometheus_exporter =
+      opentelemetry::exporter::metrics::PrometheusExporterFactory::Create(opts);
+  auto meter_provider =
+      std::make_shared<opentelemetry::sdk::metrics::MeterProvider>();
+  meter_provider->AddMetricReader(std::move(prometheus_exporter));
+  auto observability = grpc::experimental::CsmObservabilityBuilder()
+                           .SetMeterProvider(std::move(meter_provider))
+                           .BuildAndRegister();
+  if (!observability.ok()) {
+    std::cerr << "CsmObservability::Init() failed: "
+              << observability.status().ToString() << std::endl;
+    return static_cast<int>(observability.status().code());
+  }
+  GreeterClient greeter(grpc::CreateChannel(
+      absl::GetFlag(FLAGS_target),
+      absl::GetFlag(FLAGS_secure)
+          ? grpc::XdsCredentials(grpc::InsecureChannelCredentials())
```

Comment:
If the server doesn't support security, do we want to support XdsCreds on the client side?

Reply:
Done.